src/icrar/leap-accelerate/MetaData.cc

```diff
@@ -167,16 +175,46 @@ namespace icrar
      * 
      * @param metadata 
      */
-    void SetWv(MetaData& metadata)
+    void MetaData::SetWv()
     {
+        channel_wavelength = range(
+            freq_start_hz,
+            freq_start_hz + freq_inc_hz * channels,
+            freq_inc_hz);
+        
         double speed_of_light = 299792458.0;
-        metadata.channel_wavelength = range(
-            metadata.freq_start_hz,
-            metadata.freq_inc_hz,
-            metadata.freq_start_hz + metadata.freq_inc_hz * metadata.channels);
-        for(double& v : metadata.channel_wavelength)
+        for(double& v : channel_wavelength)
         {
             v = speed_of_light / v;
         }
     }
+
+    bool MetaData::operator==(const MetaData& rhs) const
```

Comment:
My reading is that this is required because `avg_data` is stored inside MetaData, but maybe it would be a better idea to store that separately and leave `MetaData` simpler? The fact that you have averaged *data* inside the *metadata* class also suggests things are getting mixed. No need to address it now though, I just wanted to share my thoughts.

Reply:
I'm also not a huge fan a MetaData class and hope to split the uninitialized data out from it. In  a later PR I move this into a `casalib` namespace to keep as a direct port from the python and introduce new a container (with no optional types) in namespace `cpu`